test/cli/other_test.py

```diff
@@ -2229,3 +2230,20 @@ def test_ignore_project_2(tmpdir):
     exitcode, stdout, _ = cppcheck(args, cwd=tmpdir)
     assert exitcode == 1, stdout
     assert stdout.splitlines() == lines_exp
+
+def test_xml_checkers_report(tmpdir):
+    test_file = os.path.join(tmpdir, 'test.c')
+    test_xml = os.path.join(tmpdir, 'results.xml')
+    with open(test_file, 'wt') as f:
```

Comment:
I would put this in helloworld_test.py, you can use the ready made testfiles instead of creating a temp folder.

and I wouldn't write the results to the file, instead let the xml be written to stderr.